eval-bot/src/main/scala/sbot/ebot/BotApi.scala

```diff
@@ -0,0 +1,180 @@
+/*
+ * Scala Bot [eval-bot]
+ *  … yet another 🤖 framework, for Scala
+ */
+
+package sbot.ebot
+
+import sbot.common.minidef._
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.MILLISECONDS
+import scala.io.StdIn
+
+import sbot.slack.api.rtm._
+import sbot.slack.api.web._
+
+import cats.~>
+import cats.data.Coproduct
+import cats.free.Free
+import cats.free.Inject
+
+import fs2.Strategy
+import fs2.util.Async
+
+object EvalApi extends EvalApiTypes {
+
+  import fs2.Task
+  import sbot.common.http.HttpPie
+  import scala.collection.immutable.List
+
+  class EvalOps[F[_]](implicit I: Inject[EvalOp, F]) {
+    import EvalOp._
+
+    def eval(code: String): Free[F, EvalResponse] = Free.inject(Eval(code))
+  }
+
+  object EvalOps {
+    def freeIn[F[_]](implicit I: Inject[EvalOp, F]): EvalOps[F] = new EvalOps[F]
+  }
+
+  sealed trait EvalOp[A]
+  object EvalOp {
+    case class Eval(code: String) extends EvalOp[EvalResponse]
+
+    def defaultInterpreter(
+      http: HttpPie,
+      evaluatorUri: String,
+      evaluatorToken: String) =
+      new (EvalOp ~> Task) {
+        def apply[A](i: EvalOp[A]) = i match {
+          case Eval(code) ⇒
+
+            http.post[EvalRequest, EvalResponse](
+              evaluatorUri,
+              EvalRequest(
+                code = code),
+              List (
+                "x-scala-eval-api-token" → evaluatorToken))
+        }
+      }
+
+  }
+
+}
+
+sealed trait EvalApiTypes {
```

Comment:
This is the same as in the evaluator client already available. We should get rid of all the manual HTTP wiring and use that instead so we can detect bugs and improve it. https://github.com/scala-exercises/scala-exercises/pull/571/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR125